src/CMakeLists.txt

```diff
@@ -83,121 +83,69 @@ set(COORDINATE_TYPE UniformCartesian) # TODO: Make this an option when more are
-  interface/container.cpp
-  interface/container.hpp
-  interface/metadata.cpp
-  interface/metadata.hpp
-  interface/params.hpp
-  interface/properties_interface.cpp
-  interface/properties_interface.hpp
-  interface/sparse_variable.cpp
-  interface/sparse_variable.hpp
-  interface/update.cpp
-  interface/update.hpp
-  interface/variable_pack.hpp
-  interface/variable.cpp
-  interface/variable.hpp
-
-  mesh/amr_loadbalance.cpp
-  mesh/domain.hpp
-  mesh/mesh_refinement.cpp
-  mesh/mesh_refinement.hpp
-  mesh/mesh.cpp
-  mesh/mesh.hpp
-  mesh/meshblock_tree.cpp
-  mesh/meshblock_tree.hpp
-  mesh/meshblock.cpp
-  mesh/weighted_ave.cpp
-
-  outputs/formatted_table.cpp
-  outputs/history.cpp
-  outputs/io_wrapper.cpp
-  outputs/io_wrapper.hpp
-  outputs/outputs.cpp
-  outputs/outputs.hpp
-  outputs/parthenon_hdf5.cpp
-  outputs/restart.cpp
-  outputs/vtk.cpp
-
-  parthenon/driver.hpp
-  parthenon/package.hpp
-  parthenon/parthenon.hpp
-  parthenon/prelude.hpp
-
-  pgen/default_pgen.cpp
-
-  reconstruct/characteristic.cpp
-  reconstruct/dc_inline.hpp
-  reconstruct/dc.cpp
-  reconstruct/plm_inline.hpp
-  reconstruct/plm.cpp
-  reconstruct/ppm.cpp
-  reconstruct/reconstruction.hpp
-  reconstruct/reconstruction.cpp
-
-  refinement/amr_criteria.cpp
-  refinement/amr_criteria.hpp
-  refinement/refinement.cpp
-  refinement/refinement.hpp
-
-  tasks/task_id.cpp
-  tasks/task_id.hpp
-  tasks/task_list.hpp
-  tasks/task_types.hpp
-
-  utils/buffer_utils.cpp
-  utils/buffer_utils.hpp
-  utils/change_rundir.cpp
-  utils/error_checking.hpp
-  utils/show_config.cpp
-  utils/signal_handler.cpp
-  utils/trim_string.cpp
-  utils/trim_string.hpp
-  utils/utils.hpp
-
-  argument_parser.hpp
-  basic_types.hpp
-  defs.hpp
-  globals.cpp
-  globals.hpp
-  kokkos_abstraction.hpp
-  parameter_input.cpp
-  parameter_input.hpp
-  parthenon_manager.cpp
-  parthenon_manager.hpp
-  parthenon_mpi.hpp
-)
+file( GLOB base_src *.cpp )
+file( GLOB base_hdr *.hpp )
+file( GLOB bvals_src bvals/*.cpp bvals/fc/*.cpp bvals/cc/*.cpp)
+file( GLOB bvals_hdr bvals/*.hpp bvals/fc/*.hpp bvals/cc/*.hpp)
+file( GLOB coordinates_src coordinates/*.cpp )
+file( GLOB coordinates_hdr coordinates/*.hpp )
+file( GLOB driver_src driver/*.cpp )
+file( GLOB driver_hdr driver/*.hpp )
+file( GLOB interface_src interface/*.cpp )
+file( GLOB interface_hdr interface/*.hpp )
+file( GLOB mesh_src mesh/*.cpp )
+file( GLOB mesh_hdr mesh/*.hpp )
+file( GLOB outputs_src outputs/*.cpp )
+file( GLOB outputs_hdr outputs/*.hpp )
+file( GLOB parthenon_src parthenon/*.cpp )
+file( GLOB parthenon_hdr parthenon/*.hpp )
+file( GLOB pgen_src pgen/*.cpp )
+file( GLOB pgen_hdr pgen/*.hpp )
+file( GLOB reconstruct_src reconstruct/*.cpp )
+file( GLOB reconstruct_hdr reconstruct/*.hpp )
+file( GLOB refinement_src refinement/*.cpp )
+file( GLOB refinement_hdr refinement/*.hpp )
+file( GLOB tasks_src tasks/*.cpp )
+file( GLOB tasks_hdr tasks/*.hpp )
+file( GLOB utils_src utils/*.cpp )
+file( GLOB utils_hdr utils/*.hpp )
```

Comment:
You can replace all the header globs easily and safely by instead using 

```
target_include_directories(parthenon PRIVATE .)
```

Reply:
@junghans I get confused by this, what is best practice for adding internal headers to library vs public?

Reply:
Not sure what you mean, this usually done by setting include directories.

Reply:
And I don’t think one needs to add the headers to the source.

Reply:
I want the header files to be attached to the library so they can be picked up for linting. Otherwise we'd have to rely on globs.

Reply:
Works great thanks @JoshuaSBrown. Header files still being linted.